homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -156,3 +207,34 @@ async def async_update(self):
         except DeviceException as ex:
             self._available = False
             _LOGGER.error("Got exception while fetching the state: %s", ex)
+
+
+class XiaomiGatewaySensor(XiaomiGatewayDevice):
+    """Representation of a XiaomiGatewaySensor."""
+
+    def __init__(self, subdevice, data_key):
+        """Initialize the XiaomiSensor."""
+        XiaomiGatewayDevice.__init__(self, subdevice)
+        self._data_key = data_key
+        self._unique_id = f"{subdevice.sid}-{data_key}"
+        self._name = f"{subdevice.sid}-{data_key}"
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        return GATEWAY_SENSOR_TYPES.get(self._data_key).icon
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return GATEWAY_SENSOR_TYPES.get(self._data_key).unit
+
+    @property
+    def device_class(self):
+        """Return the device class of this entity."""
+        return GATEWAY_SENSOR_TYPES.get(self._data_key).device_class
```

Comment:
We can't use `dict.get` and directly access the attribute of the return value. `dict.get` returns `None` if the key is missing. That would raise an `AttributeError` error now. If we know that the key is in the dict we should do `dict[key]`.
```suggestion
        return GATEWAY_SENSOR_TYPES[self._data_key].icon

    @property
    def unit_of_measurement(self):
        """Return the unit of measurement of this entity, if any."""
        return GATEWAY_SENSOR_TYPES[self._data_key].unit

    @property
    def device_class(self):
        """Return the device class of this entity."""
        return GATEWAY_SENSOR_TYPES[self._data_key].device_class
```

Reply:
You are right, thanks!
We indeed know that the key is in the dict.